sidechain/consensus/common/src/block_import.rs

```diff
@@ -83,7 +93,24 @@ where
 			state.set_last_block(block_import_params.block());
 
 			Ok(state)
-		})
+		})?;
+
+		// If the block has been proposed by this enclave, remove all successfully applied
+		// trusted calls from the top pool.
+		if block_author_is_equal_to_self::<SB, Self::Context>(ctx, sidechain_block.block_author())?
+		{
+			remove_calls_from_top_pool::<PB, SB, TopPoolExecutor>(
+				top_pool_executor,
+				sidechain_block.signed_top_hashes(),
+				&shard,
+			)
+		}
+
+		// Store block in storage.
+		// FIXME: Does this belong here?
```

Comment:
I didn't find any better place to place this ocall to update the sidechianblock storage. If you have a better solution - I'm all ears.

Reply:
Good input - I'll put that into action right away :)

Reply:
Done by @mullefel with `get_context`, Thanks alot :)